src/tslint-cli.ts

```diff
@@ -153,11 +153,24 @@ const processFile = (file: string) => {
         process.exit(1);
     }
 
+    const rulesDirectories: Array<string> = [];
+    if (configuration.rulesDirectory) {
+        if (typeof configuration.rulesDirectory === "string") {
+            rulesDirectories.push(configuration.rulesDirectory);
+        } else {
+            rulesDirectories.push(...configuration.rulesDirectory);
+        }
+    }
+
+    if (argv.r) {
+        rulesDirectories.push(argv.r);
+    }
+
     const linter = new Linter(file, contents, {
         configuration: configuration,
         formatter: argv.t,
         formattersDirectory: argv.s,
-        rulesDirectory: argv.r
+        rulesDirectory: rulesDirectories
```

Comment:
I won't make this breaking change, since it's pretty simple to keep backwards compatibility (only thing needed is calling `Linter.getRulesDirectories` to normalize the rulesDirectory argument in all places).

The `string | string[]` refactor, although, is there.